lib/location.ts

```diff
@@ -95,11 +95,11 @@ export class Location extends ReplaySubject<LocationChange>{
    * before normalizing. This method will also add a hash if `HashLocationStrategy` is
    * used, or the `APP_BASE_HREF` if the `PathLocationStrategy` is in use.
    */
-  prepareExternalUrl(url: string): string {
+  prepareExternalUrl(url: string, query: any = ''): string {
     if (url.length > 0 && !url.startsWith('/')) {
       url = '/' + url;
     }
-    return this.platformStrategy.prepareExternalUrl(url);
+    return this.platformStrategy.prepareExternalUrl(`${url}${normalizeQuery(query)}`);
```

Comment:
You'll actually want to join the strings using [`normalizeQueryParams`](https://github.com/angular/angular/blob/master/modules/angular2/src/router/location/location_strategy.ts#L63)

It basically determines if there needs to be a question mark between them. You could probably just re-implement.